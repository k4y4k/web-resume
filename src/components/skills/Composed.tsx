import 'twin.macro'
import * as React from 'react'
import Bucket from './Bucket'
import Title from './Title'

interface ComposedTypes {
  heading: string
  bucket: string[]
}

const Composed = ({ heading, bucket }: ComposedTypes): JSX.Element => (
  <div tw='mx-6 mb-4 p-4'>
    <Title category={heading} />
    <Bucket skills={bucket} />
  </div>
)

export default Composed
