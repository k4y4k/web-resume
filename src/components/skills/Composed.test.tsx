import * as React from 'react'
import { render, screen } from '@testing-library/react'
import Composed from './Composed'

describe('<Composed />', () => {
  test('Composes heading and bucket', () => {
    render(<Composed heading='Wow!' bucket={['Things!']} />)

    const bucket = screen.getByTestId('skillsBucket')
    expect(bucket).toBeInTheDocument()

    expect(screen.getByTestId('skillsComposed')).toMatchSnapshot()
  })

  describe(' incomplete data', () => {
    test('empty everything', () => {
      render(<Composed heading='' bucket={[]} />)

      const skills = screen.queryByTestId('skillsComposed')
      expect(skills).not.toBeInTheDocument()
      expect(skills).toMatchSnapshot()
    })

    test('title; bucket as empty array', () => {
      render(<Composed heading='Wow!' bucket={[]} />)

      const skills = screen.queryByTestId('skillsComposed')
      expect(skills).not.toBeInTheDocument()
      expect(skills).toMatchSnapshot()
    })

    test('title; bucket as empty string', () => {
      render(<Composed heading='' bucket={['']} />)

      const skills = screen.queryByTestId('skillsComposed')
      expect(skills).not.toBeInTheDocument()
      expect(skills).toMatchSnapshot()
    })

    test('no title; bucket as empty string', () => {
      render(<Composed heading='' bucket={['']} />)

      const skills = screen.queryByTestId('skillsComposed')
      expect(skills).not.toBeInTheDocument()
      expect(skills).toMatchSnapshot()
    })

    test('no title; bucket', () => {
      render(<Composed heading='' bucket={['Things!']} />)

      const skills = screen.queryByTestId('skillsComposed')
      expect(skills).not.toBeInTheDocument()
      expect(skills).toMatchSnapshot()
    })
  })
})
