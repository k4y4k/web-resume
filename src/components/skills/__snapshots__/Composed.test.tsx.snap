// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Composed />  incomplete data empty everything 1`] = `null`;

exports[`<Composed />  incomplete data no title; bucket 1`] = `null`;

exports[`<Composed />  incomplete data no title; bucket as empty string 1`] = `null`;

exports[`<Composed />  incomplete data title; bucket as empty array 1`] = `null`;

exports[`<Composed />  incomplete data title; bucket as empty string 1`] = `null`;

exports[`<Composed /> Composes heading and bucket 1`] = `
.c1 {
  margin-bottom: 0.25rem;
  font-weight: 700;
  font-size: 1.125rem;
  line-height: 1.25rem;
}

.c0 {
  margin-top: 0.5rem;
  margin-bottom: 0.5rem;
}

<div
  class="c0"
  data-testid="skillsComposed"
>
  <h1
    class="c1"
    data-testid="itemTitle"
  >
    Wow!
  </h1>
  <div
    data-testid="skillsBucket"
  >
    <p
      style="margin-right: 0.66rem; margin-bottom: 0.2rem; display: inline-block; line-height: 1.15;"
    >
      Things!
    </p>
  </div>
</div>
`;
